Mathlib/Combinatorics/SimpleGraph/Subgraph.lean

```diff
@@ -1187,6 +1193,17 @@ theorem deleteVerts_inter_verts_set_right_eq :
     G'.deleteVerts (s ∩ G'.verts) = G'.deleteVerts s := by
   ext <;> simp +contextual [imp_false]
 
+instance instDecidableRel_deleteVerts_adj (u : Set V) [r : DecidableRel G.Adj] :
+    DecidableRel (((⊤ : G.Subgraph).deleteVerts u).coe).Adj := by
+  intro x y
+  cases' r x y with hl hr
+  · left
+    intro hadj
+    exact hl (Subgraph.coe_adj_sub _ _ _ hadj)
+  · right
+    refine SimpleGraph.Subgraph.Adj.coe (Subgraph.deleteVerts_adj.mpr ?_)
+    exact ⟨by trivial, x.2.2, by trivial, y.2.2, hr⟩
```

Comment:
```suggestion
  cases r x y with
  | isFalse hFalse =>
    exact Decidable.isFalse (hFalse <| Subgraph.coe_adj_sub _ _ _ ·)
  | isTrue hTrue => 
    apply Decidable.isTrue
    refine SimpleGraph.Subgraph.Adj.coe (Subgraph.deleteVerts_adj.mpr ?_)
    exact ⟨trivial, x.2.2, trivial, y.2.2, hTrue⟩
```
just a suggestion: I find the explicit constructors to have clearer names than `left` and `right`.

Reply:
I think even better would be ```
... := fun x y =>
  if h : G.Adj x y 
    then _
    else _
```
so you don't need to name `r` at all. But this is also optional.